src/bindgen/ctyperesolver.rs

```diff
@@ -0,0 +1,62 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use std::collections::HashSet;
+
+pub struct CTypeResolver {
+        structs: HashSet<String>,
+        enums: HashSet<String>,
+        unions: HashSet<String>,
+}
+
+#[derive(Clone, Copy, PartialEq, PartialOrd, Eq, Ord, Debug, Hash)]
```

Comment:
A little bikesheddy here, but would you be willing to rename CType to DeclarationType? I think it's an easier name and fits the style better. And CTypeResolver to DeclarationTypeResolver.

I know there is CDeclarator inside cdecl.rs, but that's an internal type for just that module.

Reply:
Done!